docs/writing-docs/mdx.md

```diff
@@ -0,0 +1,218 @@
+// Badge.stories.mdx
+
+import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
+
+import { Badge } from './Badge';
+import { Icon } from './Icon';
+
+<Meta title="MDX/Badge" component={Badge} />
+export const Template = (args) => <Badge {...args } />
+
+# Badge
+
+Let's define a story for our `Badge` component:
+
+<Story name="positive" args={{
+    status: 'positive', 
+    label: 'Positive' 
+}}>
+  {Template.bind({})}
+</Story>
+
+We can drop it in a `Preview` to get a code snippet:
+
+<Preview>
+  <Story name="negative" args={{
+      status: 'negative', 
+      label: 'Negative'
+  }}>
+    {Template.bind({})}
+  </Story>
+</Preview>
+
+We can even preview multiple Stories in a block. This
+gets rendered as a group, but defines individual stories
+with unique URLs which is great for review and testing.
+
+<Preview>
+  <Story name="warning" args={{
+      status: warning,
+      label: 'Warning' 
+  }}>
+    {Template.bind({})}
+  </Story>
+  <Story name="neutral" args={{
+      status: 'neutral', 
+      label: 'Neutral' 
+  }}>
+    {Template.bind({})}
+  </Story>
+  <Story name="error" args={{
+      status: 'error', 
+      label: 'Error' 
+  }}>
+    {Template.bind({})}
+  </Story>
+  <Story name="with icon" args={{
+    status: warning, 
+    label: (<Icon icon="check" inline /> with icon)
+  )}}>
+    {Template.bind({})}
+  </Story>
+</Story>
+</Preview>
+```
+
+And here's how that gets rendered in Storybook:
+
+<div style="background-color:#F8FAFC">
+TODO: add image per screenshot requirements (image to match the code showed (usage of the badge component))
+</div>
+
+
+### Embedding stories
+
+Suppose you have an existing story and want to embed it into your docs. Here's how to show a story with ID some--id. Check the browser URL in Storybook v5+ to find a story's ID.
+
+```js
+
+import { Story } from "@storybook/addon-docs/blocks";
+
+# Some header
+
+And markdown here
+
+<Story id="some--id" />
+```
+
+You can also use the rest of the MDX features in conjunction with embedding. That includes source, preview, and prop tables.
+
+### Decorators and parameters
+
+To add decorators and parameters in MDX:
+
+```js
+
+<Meta
+  title='MyComponent'
+  decorators={[ ... ]}
+  parameters={{ ... }}
+/>
+
+<Story name="story" decorators={[ ... ]} parameters={{ ... }} >
+...
+</Story>
+
+```
+
+Global parameters and decorators work just like before.
+
+### Documentation-only MDX
+
+Typically, when you use Storybook MDX, you define stories in the MDX and documentation is automatically associated with those stories. But what if you want to write Markdown-style documentation without any stories inside?
+
+If you don't define stories in your MDX, you can write MDX documentation and associate it with an existing story, or embed that MDX as its own documentation node in your Storybook's navigation.
+
+<div style="background-color:#F8FAFC">
+TODO: ask tom about the "CSF Stories with MDX Docs" link if that's where it should be pointing
```

Comment:
I'm not sure. @shilman?

Reply:
https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/recipes.md#csf-stories-with-mdx-docs

This is the section. I'm not sure where this will be located after the re-org. In ADVANCED.md? recipes.md? Someplace in the official docs?

Reply:
I guess this didn't make it to these docs after all huh?. WDYT @shilman?

Reply:
Sorry I missed this. I think recipes could be its own chapter in the "Writing Docs" section, after "Doc Blocks". WDYT?

Reply:
Sure, that could make sense. OTOH I think we had previously decided not to include them, and perhaps to have a `RECIPES.md` file in the addon. I don't have a strong opinion either way.